chainercv/utils/iterator/apply_prediction_to_iterator.py

```diff
@@ -0,0 +1,141 @@
+from chainercv.utils.iterator.unzip import unzip
+
+
+def apply_prediction_to_iterator(predict, iterator, hook=None):
+    """Apply a prediction function/method to an iterator.
+
+    This function applies a prediction function/method to an iterator.
+    It assumes that the iterator returns a batch of images or
+    a batch of tuples whose first element is an image. In the case that
+    it returns a batch of tuples, the rests are treated as ground truth
+    values.
+
+    >>> imgs = next(iterator)
+    >>> # imgs: [img]
+    or
+    >>> batch = next(iterator)
+    >>> # batch: [(img, gt_val0, gt_val1)]
+
+    This function applys :func:`predict` to a batch of images and gets
+    predicted value(s). :func:`predict` should take a batch of images and
+    return a batch of prediction values
+    or a tuple of batches of prediction values.
+
+    >>> pred_vals0 = predict(imgs)
+    >>> # pred_vals0: [pred_val0]
+    or
+    >>> pred_vals0, pred_vals1 = predict(imgs)
+    >>> # pred_vals0: [pred_val0]
+    >>> # pred_vals1: [pred_val1]
+
+    Here is an exmple which applies a pretrained Faster R-CNN to
+    PASCAL VOC dataset.
+
+    >>> from chainer import iterators
+    >>>
+    >>> from chainercv.datasets import VOCDetectionDataset
+    >>> from chainercv.links import FasterRCNNVGG16
+    >>> from chainercv.utils import apply_prediction_to_iterator
+    >>>
+    >>> dataset = VOCDetectionDataset(year='2007', split='test')
+    >>> # next(iterator) -> [(img, gt_bbox, gt_label)]
+    >>> iterator = iterators.SerialIterator(
+    ...     dataset, 32, repeat=False, shuffle=False)
```

Comment:
I think it is better to set `batchsize` more than 1 because we can show `apply_prediction_to_iterator` flattens batches.